nni/nas/oneshot/pytorch/supermodule/operation.py

```diff
@@ -673,34 +701,87 @@ def forward_with_args(
             k_proj = _S(k_proj)[:, :_W(kdim)]
             v_proj = _S(cast(Tensor, self.v_proj_weight))[:embed_dim_]
             v_proj = _S(v_proj)[:, :_W(vdim)]
+        else:
+            q_proj = k_proj = v_proj = None
+
+        return in_proj_bias, in_proj_weight, bias_k, bias_v, out_proj_weight, out_proj_bias, q_proj, k_proj, v_proj, qkv_same_embed_dim
+
+    def _slice_params_mapping(self):
+        kwargs = {name: self.forward_argument(name) for name in self.argument_list}
+        in_proj_bias, in_proj_weight, bias_k, bias_v, \
+            out_proj_weight, out_proj_bias, q_proj, k_proj, v_proj, _ = self._slice_param(**kwargs)
+        return {
+            "in_proj_bias": in_proj_bias, "in_proj_weight": in_proj_weight,
+            "bias_k": bias_k, "bias_v": bias_v,
+            "out_proj.weight": out_proj_weight, "out_proj.bias": out_proj_bias,
+            "q_proj_weight": q_proj, "k_proj_weight": k_proj, "v_proj_weight": v_proj
+        }
+
+    def _save_to_sub_state_dict(self, destination, prefix, keep_vars):
```

Comment:
Why does mhattn need a dedicated `_save_to_sub_state_dict`?

Reply:
The _save_to_sub_state_dict wll save the state_dict of the current module and recursively call sub_state_dict
For mhattn , it has a sub-module `torch.nn.modules.linear.NonDynamicallyQuantizableLinear` which has no corresponding mixedop, also the mixedmhattn will process NonDynamicallyQuantizableLinear and named it "out_proj.weight".

Reply:
Please add the explanation in code comments.